ysh/funcs_builtin.py

```diff
@@ -192,6 +195,39 @@ def Init3(mem, config_parser, eval_to_dict, block_as_str, hay_result):
     SetGlobalFunc(mem, 'block_as_str', block_as_str.Call)
 
 
+class _Bool(vm._Func):
+
+    def __init__(self):
+        # type: () -> None
+        vm._Func.__init__(self)
+
+    def Run(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+        assert len(pos_args) == 1
+        assert len(named_args) == 0
+        return value.Bool(expr_eval.ToBool(pos_args[0]))
+
+
+class _StrStartsWith(vm._Func):
+
+    def __init__(self):
+        # type: () -> None
+        vm._Func.__init__(self)
+
+    def Run(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+        assert len(pos_args) == 2
+        assert len(named_args) == 0
+
+        UP_s = pos_args[0]
+        UP_prefix = pos_args[1]
+        assert UP_s.tag() == value_e.Str
```

Comment:
Another issue is what to do with the argument errors

Python tends to raise `TypeError`, so I guess we can use `raise error.InvalidType()` ?

It is going to get pretty verbose so yeah we may end up with something like Python's ParseArgs()

Right now it does stuff like

```
$ bin/ysh -c '= 1->startswith("foo")'
  = 1->startswith("foo")
  ^
[ -c flag ]:1: fatal: Expression eval error: 'int' object has no attribute 'startswith'
andy@lenny:~/git/oilshell/oil$ bin/ysh -c '= 1->startswith("foo")'
  = 1->startswith("foo")
  ^
[ -c flag ]:1: fatal: Expression eval error: 'int' object has no attribute 'startswith'
andy@lenny:~/git/oilshell/oil$ bin/ysh -c '= "foo"->startswith("foo")'
(Bool)   True
andy@lenny:~/git/oilshell/oil$ bin/ysh -c '= "foo"->startswith()'
  = "foo"->startswith()
  ^
[ -c flag ]:1: fatal: Type error in expression: startswith() takes at least 1 argument (0 given)

```